dev/build-tracker/slack.go

```diff
@@ -0,0 +1,244 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/google/go-github/v41/github"
+	"github.com/slack-go/slack"
+	"github.com/sourcegraph/log"
+	"github.com/sourcegraph/sourcegraph/dev/team"
+)
+
+type NotificationClient struct {
+	slack   slack.Client
+	team    team.TeammateResolver
+	logger  log.Logger
+	Channel string
+}
+
+func NewNotificationClient(logger log.Logger, slackToken, githubToken, channel string) *NotificationClient {
+	slack := slack.New(slackToken)
+
+	httpClient := http.Client{
+		Timeout: 5 * time.Second,
+	}
+	githubClient := github.NewClient(&httpClient)
+	teamResolver := team.NewTeammateResolver(githubClient, slack)
+
+	return &NotificationClient{
+		logger:  logger.Scoped("notificationClient", "client which interacts with Slack and Github to send notifications"),
+		slack:   *slack,
+		team:    teamResolver,
+		Channel: channel,
+	}
+}
+
+func (c *NotificationClient) getTeammateForBuild(build *Build) (*team.Teammate, error) {
+	if build.Author == nil {
+		return nil, errors.New("nil Author")
+	}
+	return c.team.ResolveByCommitAuthor(context.Background(), "sourcegraph", "sourcegraph", build.commit())
+}
+
+func (c *NotificationClient) sendNotification(build *Build) error {
```

Comment:
```suggestion
func (c *NotificationClient) send(build *Build) error {
```

we know it's a notification as we are on that type